### PR TITLE
Prevent crash with `getContexts` and fix the window reference in `reportError`

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Avoids calling the broken chrome api `runtime.getContexts()` that is crashing the browser
- Fixes a `window` reference in the `reportError` code chain that is now being executed in the service worker, which does not have `window` available

_Example debug logs showing behavior for two errors in a row_:

image

_Example logs for two errors in quick succession to force the promise concurrency handling_:

image

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

**Comment by BLoe** [comment]
Note: If we update this branch with `main`, we will no longer be able to test this with the `sidePanel.open()` error (which is fixed on `main`)

File: src/data/service/errorService.ts [hunk 1]
```diff
@@ -162,7 +162,7 @@ export async function reportToErrorService(
     extension_uuid: flatContext.modComponentId,
     extension_label: flatContext.modComponentLabel,
     step_label: flatContext.label,
-    user_agent: window.navigator.userAgent,
+    user_agent: navigator.userAgent,
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/data/service/errorService.ts:165]: Should this be `globalThis?.navigator?.userAgent`?

**Comment by twschiller** [comment]
@grahamlangford @BLoe should we be adding a kill switch to prevent trying to send to DataDog?

**Comment by grahamlangford** [comment]
> @grahamlangford @BLoe should we be adding a kill switch to prevent trying to send to DataDog?

Let's do that in a follow-on. Doesn't hurt to have as a just-in-case, but I want to get this merged now so I can get it into beta.3

File: src/data/service/errorService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (BLoe) [reply src/data/service/errorService.ts:165]: It seems like it works? I'm not sure 🤷‍♂️ 
image

[the remark this answers is not in the document]
Comment (grahamlangford) [reply src/data/service/errorService.ts:165]: See: https://stackoverflow.com/questions/73778202/using-window-globals-in-manifestv3-service-worker-background-script

**Review by grahamlangford — APPROVED** [review]

File: src/data/service/errorService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (BLoe) [reply src/data/service/errorService.ts:165]: See this accepted SO response: https://stackoverflow.com/questions/73778202/using-window-globals-in-manifestv3-service-worker-background-script

**Pull request merged**